spark-cassandra-connector/src/it/scala/com/datastax/spark/connector/rdd/RDDSpec.scala

```diff
@@ -201,6 +201,38 @@ class RDDSpec extends SparkCassandraITFlatSpecBase {
     checkArrayCassandraRow(result)
   }
 
+  it should "throw a meaningful exception if partition column is null when joining with Cassandra table" in {
+    val source = sc.parallelize(keys).map(x ⇒ new KVWithOptionRow(None))
+    try {
+      source.joinWithCassandraTable[(Int, Long, String)](ks, tableName).collect()
+      fail()
```

Comment:
As you can see, I need to verify the exception message - that is why I did it in this way